module/Finna/src/Finna/RecordDriver/Feature/SolrFinnaTrait.php

```diff
@@ -1175,7 +1175,11 @@ protected function resolveUrlTypes($urls)
      */
     protected function getSupportedCitationFormats()
     {
-        return ['APA', 'Chicago', 'MLA', 'Harvard'];
+        $supportedFormats = ['APA', 'Chicago', 'MLA', 'Harvard'];
+        if (isset($this->fields['hierarchy_top_id']) && str_starts_with($this->getRecordFormat(), 'ead')) {
```

Comment:
Olisi parempi tarkistaa nykyinen luokka, esim. `$this instanceof SolrEad`, mutta siisteintä olisi lisätä getSupportedCitationFormats-metodi SolrEad-luokkaan (vai SolrEad3?).